futurex_openedx_extensions/dashboard/statistics/courses.py

```diff
@@ -208,41 +212,68 @@ def get_courses_count_by_status(
     return q_set
 
 
+def _cache_key_courses_ratings(tenant_id: int, visible_filter: bool | None, active_filter: bool | None) -> str:
+    """
+    Generate cache key for get_courses_ratings
+
+    :param tenant_id: Tenant ID
+    :type tenant_id: int
+    :param visible_filter: Value to filter courses on catalog visibility
+    :type visible_filter: bool | None
+    :param active_filter: Value to filter courses on active status
+    :type active_filter: bool | None
+    :return: Cache key string
+    :rtype: str
+    """
+    return f'fx_courses_ratings_t{tenant_id}_v{visible_filter}_a{active_filter}'
+
+
 def get_courses_ratings(
-    fx_permission_info: dict,
+    tenant_id: int,
     visible_filter: bool | None = True,
     active_filter: bool | None = None,
 ) -> Dict[str, int]:
     """
-    Get the average rating of courses in the given tenants
+    Get the average rating of courses for a single tenant. Results are cached per tenant.
 
-    :param fx_permission_info: Dictionary containing permission information
-    :type fx_permission_info: dict
+    :param tenant_id: Tenant ID to get ratings for
+    :type tenant_id: int
     :param visible_filter: Value to filter courses on catalog visibility. None means no filter
     :type visible_filter: bool | None
     :param active_filter: Value to filter courses on active status. None means no filter (according to dates)
     :type active_filter: bool | None
     :return: Dictionary containing the total rating, courses count, and rating count per rating value
     :rtype: Dict[str, int]
     """
-    q_set = get_base_queryset_courses(
-        fx_permission_info, visible_filter=visible_filter, active_filter=active_filter
+    @cache_dict(
+        timeout='FX_CACHE_TIMEOUT_COURSES_RATINGS',
+        key_generator_or_name=_cache_key_courses_ratings
     )
+    def _get_ratings(t_id: int, v_filter: bool | None, a_filter: bool | None) -> Dict[str, int]:
```

Comment:
why the inner function? I don't see how that is needed

You can add the cache decorator on the original function.